src/encoding.ts

```diff
@@ -48,14 +49,14 @@ export interface UnitEncoding {
    * - For `text` – the text's font size.
    * - Size is currently unsupported for `line` and `area`.
    */
-  size?: ChannelDefWithLegend;
+  size?: LegendFieldDef | ValueDef<number>;
 
   /**
    * The symbol's shape (only for `point` marks). The supported values are
    * `"circle"` (default), `"square"`, `"cross"`, `"diamond"`, `"triangle-up"`,
    * or `"triangle-down"`, or else a custom SVG path string.
    */
-  shape?: ChannelDefWithLegend; // TODO: maybe distinguish ordinal-only
+  shape?: LegendFieldDef | ValueDef<string>; // TODO: maybe distinguish ordinal-only
```

Comment:
Well, we actually use `Shape | string` in https://github.com/vega/vega-lite/blob/f39a127d70248253c7ec4af93a1290ce9400cfd6/src/mark.ts#L203